week2/LogAnalysis.hs

```diff
@@ -0,0 +1,56 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module LogAnalysis where
+
+import Log
+
+-- EXERCISE 1 --
+
+parseMessage :: String -> LogMessage
+parseMessage = 
+  convert . words
+  where convert :: [String] -> LogMessage
+        convert ("I":t:m) = LogMessage Info (read t) (unwords m)
```

Comment:
What happens if `t` is not a number?

Reply:
It will throw an exception and crash

Reply:
Suggestion on how to catch the exception or prevent it? wrapped in Maybe?

Reply:
https://www.stackage.org/haddock/lts-13.21/base-4.12.0.0/Text-Read.html#v:readMaybe

Reply:
Or you can use [maybe](https://hackage.haskell.org/package/base-4.12.0.0/docs/Prelude.html#v:maybe) to tell the function what to do if the read fails